### PR TITLE
Run app migrations when installing from package

### LINKED ISSUE #48: DB creation migration of apps randomly not running
### Steps

1. Checkout core master
1. Delete "apps/customgroups" if any available
1. Setup OC from scratch
1. Clear cookies and cache in browser
1. Go to marketplace
1. Install "customgroups" app
1. Refresh the page

### Expected result
App enabled properly.

### Actual result
Internal server error.

```
{"reqId":"6OKjs4ULe7a6l47pujXB","level":3,"time":"2017-04-24T13:57:03+00:00","remoteAddr":"127.0.0.1","user":"admin","app":"index","method":"GET","url":"\/owncloud\/index.php\/apps\/market\/apps","message":"Exception: {\"Exception\":\"Doctrine\\\\DBAL\\\\Exception\\\\TableNotFoundException\",\"Message\":\"An exception occurred while executing 'SELECT m.`group_id`, m.`user_id`, m.`role`, g.`uri`, g.`display_name` FROM `oc_custom_group_member` m, `oc_custom_group` g WHERE (g.`group_id` = m.`group_id`) AND (`user_id` = ?) ORDER BY m.`group_id` ASC' with params [\\\"admin\\\"]:\\n\\nSQLSTATE[42S02]: Base table or view not found: 1146 Table 'owncloud.oc_custom_group_member' doesn't exist\",\"Code\":0,\"Trace\":\"#0 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/composer\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/DBALException.php(128): Doctrine\\\\DBAL\\\\Driver\\\\AbstractMySQLDriver->convertException('An exception oc...', Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOException))\\n#1 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/composer\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Connection.php(855): Doctrine\\\\DBAL\\\\DBALException::driverExceptionDuringQuery(Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOMySql\\\\Driver), Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOException), 'SELECT m.`group...', Array)\\n#2 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/DB\\\/Connection.php(190): Doctrine\\\\DBAL\\\\Connection->executeQuery('SELECT m.`group...', Array, Array, NULL)\\n#3 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/composer\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Query\\\/QueryBuilder.php(206): OC\\\\DB\\\\Connection->executeQuery('SELECT m.`group...', Array, Array)\\n#4 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/DB\\\/QueryBuilder\\\/QueryBuilder.php(141): Doctrine\\\\DBAL\\\\Query\\\\QueryBuilder->execute()\\n#5 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/apps\\\/customgroups\\\/lib\\\/CustomGroupsDatabaseHandler.php(128): OC\\\\DB\\\\QueryBuilder\\\\QueryBuilder->execute()\\n#6 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/apps\\\/customgroups\\\/lib\\\/CustomGroupsBackend.php(82): OCA\\\\CustomGroups\\\\CustomGroupsDatabaseHandler->getUserMemberships('admin', NULL)\\n#7 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/Group\\\/Manager.php(300): OCA\\\\CustomGroups\\\\CustomGroupsBackend->getUserGroups('admin')\\n#8 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/util.php(249): OC\\\\Group\\\\Manager->getUserIdGroups('admin')\\n#9 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/base.php(909): OC_Util::setupFS()\\n#10 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/index.php(49): OC::handleRequest()\\n#11 {main}\",\"File\":\"\\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/composer\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Driver\\\/AbstractMySQLDriver.php\",\"Line\":53}"}
```

The DB creation migrations didn't run correctly.

Note that if you redo the same steps but **keep** the "customgroups" folder, then it will work fine.

Happens fairly consistently.

@SergioBertolinSG I remember you observed this once.

@DeepDiver1975

### PR DESCRIPTION
**Pull request by PVince81**
## Description
Run app migrations when installing from package
See https://github.com/owncloud/market/issues/48#issuecomment-296680428 for an explanation why.

## Related Issue
Fixes https://github.com/owncloud/market/issues/48

## Motivation and Context
See https://github.com/owncloud/market/issues/48#issuecomment-296680428

## How Has This Been Tested?
Install customgroups app from market.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

Please review @DeepDiver1975 @VicDeo

### CODE REVIEW

File: lib/private/App/AppManager.php [hunk 1]
```diff
@@ -383,10 +383,13 @@ public function getAlwaysEnabledApps() {
 	 * @since 10.0
 	 */
 	public function installApp($package) {
-		return Installer::installApp([
+		$appId = Installer::installApp([
 			'source' => 'local',
 			'path' => $package
 		]);
+		// HACK: this will run the migration and related code
```
Comment (butonic) [inline lib/private/App/AppManager.php]: Can you add a TODO comment with what needs to be done to fix it properly?

Comment (PVince81) [inline lib/private/App/AppManager.php]: tbh I have no idea... kill the legacy app code ?

**Comment by VicDeo** [comment]
:+1: works

File: lib/private/App/AppManager.php (hunk 1 above)
Comment (PVince81) [inline lib/private/App/AppManager.php]: @butonic I made a separate ticket, I think it's a bigger topic: https://github.com/owncloud/core/issues/27736

**Pull request merged**